be/src/object_store/s3_object_store.cpp

```diff
@@ -301,6 +422,42 @@ Status S3ObjectStore::list_objects(const std::string& bucket, const std::string&
     return Status::OK();
 }
 
+StatusOr<std::unique_ptr<ObjectStore>> new_s3_object_store() {
+    Aws::Client::ClientConfiguration config;
+    config.scheme = Aws::Http::Scheme::HTTP;
+    config.maxConnections = config::aws_s3_max_connection;
+    if (!config::aws_s3_endpoint.empty()) {
+        config.endpointOverride = config::aws_s3_endpoint;
+    }
+    if (!config::aws_s3_region.empty()) {
+        config.region = config::aws_s3_region;
+    }
+    auto credentials = std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(config::aws_access_key_id.c_str(),
+                                                                                 config::aws_secret_access_key.c_str());
+    auto s3client = std::make_shared<Aws::S3::S3Client>(credentials, config);
+    auto store = std::make_unique<S3ObjectStore>(std::move(s3client), false);
+    return std::move(store);
+}
+
+class S3Register {
+public:
+    explicit S3Register(std::string_view pattern) {
+        ObjectStore::register_store(pattern, [](std::string_view /*uri*/) -> StatusOr<std::unique_ptr<ObjectStore>> {
+            return new_s3_object_store();
+        });
+    }
+};
+
+void s3_global_init() {
+    // Simply define these variables as global variables won't work, because object store
+    // is compiled into a static library, and global variables defined in static library will
+    // not be initialized.
+    static S3Register reg0("s3://.*");
+    static S3Register reg1("s3a://.*");
+    static S3Register reg2("s3n://.*");
+    static S3Register reg3("oss://.*");
```

Comment:
OSS shoud be registered in oss_global_init?

Reply:
OSS is a S3 compatible service.